pages/account/manage/profile.js

```diff
@@ -74,33 +75,39 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
 
   const handleSubmit = async (e) => {
     e.preventDefault();
-    const res = await fetch(`${BASE_URL}/api/account/manage/profile`, {
-      method: "PUT",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({ name, bio, tags, layout }),
-    });
-    const update = await res.json();
 
-    if (update.message) {
-      return setShowNotification({
+    try {
+      const update = await sendRequest({
+        url: `${BASE_URL}/api/account/manage/profile`,
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: { name, bio, tags, layout },
+      });
+
+      setTags(update.tags);
+      setShowNotification({
+        show: true,
+        type: "success",
+        message: "Profile updated",
+        additionalMessage: "Your profile has been updated successfully",
+      });
+    } catch (err) {
+      const errMessage =
+        typeof err === "string"
```

Comment:
Duplicate logic as in `milestone/[[...data]]` file, I think it can be moved to some function (f.e. formatError)

Reply:
Great idea👍